apps/andi/lib/andi_web/controllers/api/ingestion_controller.ex

```diff
@@ -144,9 +144,23 @@ defmodule AndiWeb.API.IngestionController do
   end
 
   defp add_uuid(message) do
-    uuid = UUID.uuid4()
-
-    Map.merge(message, %{"id" => uuid}, fn _k, v1, _v2 -> v1 end)
+    if Map.has_key?(message, "id") do
+      case IngestionStore.get(Map.get(message, "id")) do
+        {:ok, nil} ->
+          {:invalid, "Do not include id in create call"}
+
+        {:ok, _} ->
+          uuid = UUID.uuid4()
+          {:ok, Map.merge(%{"id" => uuid}, message)}
```

Comment:
If you find an existing ingestion with the provided ID, you wouldn't want to generate a new ID. This would effectively delete the old ingestion and create a new one. I believe you would want to just return the message since this has been determined as an update message with no need to add_uuid

Reply:
oop, yeah good catch 🤦